### PR TITLE
FEAT: Handle responsive login/register page for mobile (FNW-80)

### PR DESCRIPTION
**Pull request by mathieurobert1**
This pull request introduces a set of changes aimed at improving the mobile user experience for the login and signup pages in the `react-frontend` project. The changes include adding mobile-specific components and a custom hook to detect mobile devices.

### Mobile Experience Enhancements:

* **New Custom Hook for Mobile Detection:**
  - [`react-frontend/src/components/IsMobile/isMobile.tsx`](diffhunk://#diff-347c08d594c734ece364cccf36fe6fa1ce81e188f0b8e3fd713c37bf33927ca6R1-R13): Added a custom hook `useIsMobile` to determine if the user is on a mobile device. This hook listens to window resize events to update its state.

* **Login Page Adjustments:**
  - [`react-frontend/src/pages/Login/login.tsx`](diffhunk://#diff-0330cf7cc55b0e8d0467aac502084bd103ef31bb063772fe5c9cd3808b43ec6cR6-R7): Integrated the `useIsMobile` hook to conditionally render a new `MobileLogin` component for mobile users. [[1]](diffhunk://#diff-0330cf7cc55b0e8d0467aac502084bd103ef31bb063772fe5c9cd3808b43ec6cR6-R7) [[2]](diffhunk://#diff-0330cf7cc55b0e8d0467aac502084bd103ef31bb063772fe5c9cd3808b43ec6cL20-R33)
  - [`react-frontend/src/pages/Login/mobileLogin.tsx`](diffhunk://#diff-65133dc3de0cbd66132dd4d71d96de8d0bfd98693006670882c0e919ffd14eb1R1-R76): Created a new `MobileLogin` component to provide a mobile-optimized login experience.

* **Signup Page Adjustments:**
  - [`react-frontend/src/pages/Signup/signup.tsx`](diffhunk://#diff-20d286abc538bfb10860274a16444dd9955db30265dedec8a312b1ee1983f273R6-R7): Integrated the `useIsMobile` hook to conditionally render a new `MobileSignup` component for mobile users. [[1]](diffhunk://#diff-20d286abc538bfb10860274a16444dd9955db30265dedec8a312b1ee1983f273R6-R7) [[2]](diffhunk://#diff-20d286abc538bfb10860274a16444dd9955db30265dedec8a312b1ee1983f273L26-R41)
  - [`react-frontend/src/pages/Signup/mobileSignup.tsx`](diffhunk://#diff-9595b87290fa37545041c42edc5ba9b430086b0e9863117131a078ce7ca185a0R1-R83): Created a new `MobileSignup` component to provide a mobile-optimized signup experience.

### Configuration Updates:

* **Deployment Target Selector Configuration:**
  - [`react-frontend/android/.idea/deploymentTargetSelector.xml`](diffhunk://#diff-98ac9fa306d2971030f08831c5738dd72a8444deedde87b2e2684357cbdb3a02R1-R10): Added a new configuration file for deployment target selection in the Android project.

image
image

### CODE REVIEW

File: react-frontend/src/pages/Signup/signup.tsx [hunk 1]
```diff
@@ -12,18 +14,31 @@ export default function Signup() {
     const [confirmPassword, setConfirmPassword] = useState('');
     const navigate = useNavigate();
 
-    const sendSignUp = async(e: React.FormEvent) => {
+    const sendSignUp = async (e: React.FormEvent) => {
```
Comment (Peralban) [inline react-frontend/src/pages/Signup/signup.tsx:17]: a little space

File: react-frontend/src/pages/Signup/signup.tsx [hunk 2]
```diff
@@ -105,7 +120,7 @@ export default function Signup() {
                         {/* Oauth2 Login */}
                         <div className="md:w-1/6 mt-6 md:mt-0">
                             <p className="text-sl mb-4 text-customYellow text-shadow-custom font-instrumentSans text-center">
-                                or<br/>continue with
+                                or<br />continue with
```
Comment (Peralban) [inline react-frontend/src/pages/Signup/signup.tsx:123]: little space here too

**Review by Peralban — CHANGES_REQUESTED** [review]
good, but 2 typo errors ✍️

**Review by AubaneNourry — APPROVED** [review]
Good job :)
Maybe you can update the result in the figma according to the new style chosen (if you didn't already)

File: react-frontend/src/pages/Signup/signup.tsx (hunk 1 above)
Comment (mathieurobert1) [reply react-frontend/src/pages/Signup/signup.tsx:17]: See with Alban

File: react-frontend/src/pages/Signup/signup.tsx (hunk 2 above)
Comment (mathieurobert1) [reply react-frontend/src/pages/Signup/signup.tsx:123]: It's because it open and close the context, it's two different things.

**Review by Peralban — APPROVED** [review]

**Pull request merged**